src/kv/KeyValueDB.h

```diff
@@ -226,77 +246,138 @@ class KeyValueDB {
   };
   typedef ceph::shared_ptr< WholeSpaceIteratorImpl > WholeSpaceIterator;
 
+  /// Iterator for column family which is created explicitly and not the default one.
+  class ColumnFamilyIteratorImpl {
+  public:
+    virtual int seek_to_first() = 0;
+    virtual int seek_to_last() = 0;
+    virtual int upper_bound(const std::string &after) = 0;
+    virtual int lower_bound(const std::string &to) = 0;
+    virtual bool valid() = 0;
+    virtual int next() = 0;
+    virtual int prev() = 0;
+    virtual std::string key() = 0;
+    virtual bufferlist value() = 0;
+    virtual bufferptr value_as_ptr() {
+      bufferlist bl = value();
+      if (bl.length()) {
+        return *bl.buffers().begin();
+      } else {
+        return bufferptr();
+      }
+    }
+    virtual int status() = 0;
+    virtual ~ColumnFamilyIteratorImpl() { }
+  };
+  typedef ceph::shared_ptr< ColumnFamilyIteratorImpl > ColumnFamilyIterator;
+
```

Comment:
do we need to take care of any tools like ceph-kvstore-tool etc.. to work with column families. These tools are used to repair/list the entries in DB.

Reply:
@varadakari I didn't do the investigate yet ... it sounds needed ... Thanks for your remind. 

For this PR, we want to first introduce the feature of cfs, and it will be controlled by an option(false by default). We'd like to support the the tools after we get conclusion about this PR.

Reply:
Sure. Just check once and have it in the TODO list, if needed